kratos/utilities/variable_utils.h

```diff
@@ -101,6 +101,69 @@ class KRATOS_API(KRATOS_CORE) VariableUtils
     ///@name Operations
     ///@{
 
+    /**
+     * @brief Copies the nodal value of a variable from an origin model 
+     * part nodes to the nodes in a destination model part. It is assumed that 
+     * both origin and destination model parts have the same number of nodes.
+     * @param rVariable reference to the variable to be set
+     * @param rOriginModelPart origin model part from where the values are retrieved
+     * @param rDestinationModelPart destination model part to where the values are copied to
+     * @param BuffStep buffer step
+     */
+    template< class TVarType >
+    void CopyModelPartNodalVar(
+        TVarType& rVariable,
+        ModelPart& rOriginModelPart,
+        ModelPart& rDestinationModelPart,
+        const unsigned int BuffStep = 0){
+
+        auto n_orig_nodes = rOriginModelPart.NumberOfNodes();
+        auto n_dest_nodes = rDestinationModelPart.NumberOfNodes();
+
+        KRATOS_ERROR_IF_NOT(n_orig_nodes == n_dest_nodes) << "Origin and destination model parts have different number of nodes."
+                                                        << "\n\t- Number of origin nodes: " << n_orig_nodes 
+                                                        << "\n\t- Number of destination nodes: " << n_dest_nodes << std::endl;
+
+        #pragma omp parallel for
+        for(int i_node = 0; i_node < static_cast<int>(n_orig_nodes); ++i_node){
+            auto it_dest_node = rDestinationModelPart.NodesBegin() + i_node;
+            const auto it_orig_node = rOriginModelPart.NodesBegin() + i_node;
+            const auto r_value = it_orig_node->GetSolutionStepValue(rVariable, BuffStep); 
```

Comment:
I'd do `const auto&`, since you may be copying vectors (or potentially heavier things).

Reply:
Totally agree.